enhancements/insights/pulling-pull-secret-from-ocm.md

```diff
@@ -0,0 +1,188 @@
+---
+title: pulling-and-updating-pull-secret-from-ocm
+authors:
+  - "@tremes"
+reviewers:
+  - "@mrunalp"
+  - "@aravindhp"
+  - "@inecas"
+  - "@rawsyntax"
+  - "@markturansky"
+approvers:
+  - "@mrunalp"
+  - "@aravindhp" 
+creation-date: 2021-12-01
+last-updated: 2021-12-16
+status: implementable
+see-also:
+  - "https://issues.redhat.com/browse/CCXDEV-6529"
+  - "https://issues.redhat.com/browse/CCX-185"
+---
+
+# Insights Operator pulling and updating `pull-secret` from the OCM API
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+- [x] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [x] User-facing documentation is created/updated in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+This enhancement will enable the Insights Operator to pull the updated `pull-secret` (from the OCM API)
+and that will enable cluster ownership transfer. Currently, the ownership transfer is a two step process
+which involves initiating the transfer from the OpenShift Cluster Manager (OCM) API
+and then updating the global `pull-secret` on the cluster with the new owner’s `pull-secret`.
+If the `pull-secret` is not changed on the cluster within 5 days, the ownership transfer is canceled
+and needs to be initiated again.
+This process applies to all scenarios that require cluster ownership transferred across Red Hat accounts.
+
+The intent is to simplify the cluster ownership process to a one step process.
+
+## Motivation
+
+The entry point to many OpenShift subscriptions is for a user to start with a 60 day trial from the OCM with personal credentials
+and then migrate it over to a company or organization account. In addition, reports from the field indicate that customers in general when transferring ownership, either follow one of the steps in the process and do not follow through with the other step required to complete the transfer. This enhancement will make the transfer frictionless and provide a better user experience.
+
+
+### Goals
+The goal of this enhancement is to describe the API and the interaction between OCM and the cluster that is being transferred to update the `pull-secret` on that cluster:
+
+- Extend the Insights Operator config with a new OCM API endpoint to be able to query the new `pull-secret` content
+- Periodically (once a day) query the new endpoint and if there is a new or updated content then update the `pull-secret`. User can trigger the update faster by deleting the Insights operator Pod (This needs to be documented and may change in the future - e.g we can introduce a new CRD to configure this).
```

Comment:
>User can trigger the update faster by deleting the Insights operator Pod (This needs to be documented and may change in the future - e.g we can introduce a new CRD to configure this).

I suggest calling this out under `Risks and Mitigations`

Reply:
OK. Thanks. Done